src/parser/ts/index.ts

```diff
@@ -0,0 +1,525 @@
+import {
+  DataQueryRequestItem,
+  QueryOptions,
+  TimeSeriesResponseItem,
+  QueryTarget,
+  IdEither,
+} from '../../types';
+import _, { isArray, isObjectLike } from 'lodash';
+import { TemplateSrv } from 'grafana/app/features/templating/template_srv';
+import { Parser, Grammar } from 'nearley';
+import grammar from './grammar';
+import { getTimeseries, getLabelWithInjectedProps } from '../../cdfDatasource';
+import { Connector } from '../../connector';
+import { FilterType, ParsedFilter } from '../types';
+import { applyFilters } from '../../utils';
+import { parseWith } from '../events-assets/index';
+import getFilterDeep from 'deepdash/getFilterDeep';
+
+const filterDeep = getFilterDeep(_);
+const compiledGrammar = Grammar.fromCompiled(grammar);
+
+export const formQueriesForExpression = async (
+  expr: string,
+  options: QueryOptions,
+  templateSrv: TemplateSrv,
+  target: QueryTarget,
+  connector: Connector
+): Promise<DataQueryRequestItem[]> => {
+  const templatedExpr = templateSrv.replace(expr.trim(), options.scopedVars);
```

Comment:
do you really need this service here? We can replace variables before passing expr here

Reply:
good point, fixed now